### ISSUE TITLE
Call to undefined method Smile_ElasticSearch_Model_Resource_Engine_Elasticsearch_Mapping_Product::resetSearchResults()

### ISSUE THREAD

**Issue by matrixbegins** [Smile-SA/smile-magento-elasticsearch#42]
HI All
I am running version `1.1.2` of this module. I have a cron that imports products from ebay. It uses magento models and save products in DB.
 When I change the `catalog search index` mode to `update on save`  I get following error frequently.

```

 PHP Fatal error:  Call to undefined method Smile_ElasticSearch_Model_Resource_Engine_Elasticsearch_Mapping_Product::resetSearchResults() in /var/www/live-app/app/code/community/Smile/ElasticSearch/Model/Indexer/Fulltext.php on line 145
 PHP Stack trace:
 PHP   1. {main}() /var/www/live-app/index.php:0
 PHP   2. Mage::run() /var/www/live-app/index.php:87
 PHP   3. Mage_Core_Model_App->run() /var/www/live-app/app/Mage.php:683
 PHP   4. Mage_Core_Controller_Varien_Front->dispatch() /var/www/live-app/app/code/core/Mage/Core/Model/App.php:354
 PHP   5. Mage_Core_Controller_Varien_Router_Standard->match() /var/www/live-app/app/code/core/Mage/Core/Controller/Varien/Front.php:172
 PHP   6. Mage_Core_Controller_Varien_Action->dispatch() /var/www/live-app/app/code/core/Mage/Core/Controller/Varien/Router/Standard.php:250
 PHP   7. Mage_Adminhtml_Catalog_ProductController->saveAction() /var/www/live-app/app/code/core/Mage/Core/Controller/Varien/Action.php:418
 PHP   8. Mage_Core_Model_Abstract->save() /var/www/live-app/app/code/core/Mage/Adminhtml/controllers/Catalog/ProductController.php:728
 PHP   9. Mage_Catalog_Model_Product->_afterSave() /var/www/live-app/app/code/core/Mage/Core/Model/Abstract.php:319
 PHP  10. Mage_Core_Model_Abstract->_afterSave() /var/www/live-app/app/code/core/Mage/Catalog/Model/Product.php:549
 PHP  11. Mage::dispatchEvent() /var/www/live-app/app/code/core/Mage/Core/Model/Abstract.php:466
 PHP  12. Mage_Core_Model_App->dispatchEvent() /var/www/live-app/app/Mage.php:447
 PHP  13. Mage_Core_Model_App->_callObserverMethod() /var/www/live-app/app/code/core/Mage/Core/Model/App.php:1317
 PHP  14. Mage_CatalogInventory_Model_Observer->saveInventoryData() /var/www/live-app/app/code/core/Mage/Core/Model/App.php:1338
 PHP  15. Mage_Core_Model_Abstract->save() /var/www/live-app/app/code/core/Mage/CatalogInventory/Model/Observer.php:154
 PHP  16. Mage_CatalogInventory_Model_Stock_Item->_afterSave() /var/www/live-app/app/code/core/Mage/Core/Model/Abstract.php:319
 PHP  17. Mage_Index_Model_Indexer->processEntityAction() /var/www/live-app/app/code/core/Mage/CatalogInventory/Model/Stock/Item.php:792
 PHP  18. Mage::dispatchEvent() /var/www/live-app/app/code/core/Mage/Index/Model/Indexer.php:334
 PHP  19. Mage_Core_Model_App->dispatchEvent() /var/www/live-app/app/Mage.php:447
 PHP  20. Mage_Core_Model_App->_callObserverMethod() /var/www/live-app/app/code/core/Mage/Core/Model/App.php:1317
 PHP  21. Mage_CatalogInventory_Model_Observer->reindexProductsMassAction() /var/www/live-app/app/code/core/Mage/Core/Model/App.php:1338
 PHP  22. Mage_Index_Model_Indexer->indexEvents() /var/www/live-app/app/code/core/Mage/CatalogInventory/Model/Observer.php:973
 PHP  23. Mage_Index_Model_Indexer->_runAll() /var/www/live-app/app/code/core/Mage/Index/Model/Indexer.php:223
 PHP  24. call_user_func_array() /var/www/live-app/app/code/core/Mage/Index/Model/Indexer.php:378
 PHP  25. Mage_Index_Model_Process->indexEvents() /var/www/live-app/app/code/core/Mage/Index/Model/Indexer.php:378
 PHP  26. Mage_Index_Model_Process->_processEventsCollection() /var/www/live-app/app/code/core/Mage/Index/Model/Process.php:360
 PHP  27. Mage_Index_Model_Process->processEvent() /var/www/live-app/app/code/core/Mage/Index/Model/Process.php:384
 PHP  28. Smile_ElasticSearch_Model_Indexer_Fulltext->processEvent() /var/www/live-app/app/code/core/Mage/Index/Model/Process.php:282
 PHP  29. Mage_Index_Model_Indexer_Abstract->processEvent() /var/www/live-app/app/code/community/Smile/ElasticSearch/Model/Indexer/Fulltext.php:50
 PHP  30. Smile_ElasticSearch_Model_Indexer_Fulltext->_processEvent() /var/www/truefacets/app/code/core/Mage/Index/Model/Indexer/Abstract.php:103
```
Stack trace also suggest that even when saving product from backend its reports the same error.

Can anyone help here on this?

**Comment by romainruaud** [comment]
Hello,

thank you for submitting this issue. Is it still happening ? Could you precise your Magento version ?

Best regards,

**Comment by afoucret** [comment]
This one have been fixed since a while in commit 5f880fd
The fix is available in version 1.2 and superior.

I close the issue since it is fixed.
